js/ui/folderModel.js

```diff
@@ -0,0 +1,63 @@
+//-*- mode: js; js-indent-level: 4; indent-tabs-mode: nil -*-
+const Gio = imports.gi.Gio;
+const GLib = imports.gi.GLib;
+const Gtk = imports.gi.Gtk;
+
+const Lang = imports.lang;
+
+const _PREFIX = 'eos-folder';
+
+const FolderModel = new Lang.Class({
+    Name: 'FolderModel',
+
+    createFolder: function(name, iconName) {
+        let keyFile = new GLib.KeyFile();
+
+        keyFile.set_value(GLib.KEY_FILE_DESKTOP_GROUP,
+                          GLib.KEY_FILE_DESKTOP_KEY_NAME, name);
+
+        keyFile.set_value(GLib.KEY_FILE_DESKTOP_GROUP,
+                          GLib.KEY_FILE_DESKTOP_KEY_ICON, iconName);
+
+        keyFile.set_value(GLib.KEY_FILE_DESKTOP_GROUP,
+                          GLib.KEY_FILE_DESKTOP_KEY_TYPE, GLib.KEY_FILE_DESKTOP_TYPE_DIRECTORY);
+
+        let dirpath = GLib.build_filenamev([GLib.get_user_data_dir(), '/desktop-directories']);
+
+        if (GLib.mkdir_with_parents(dirpath, parseInt('0755', 8), null) < 0) {
+            log('could not create the directory ' + dirpath);
+            return null;
+        }
+
+        let dir =  Gio.File.new_for_path(dirpath);
+        let enumerator = dir.enumerate_children('standard::name',
+                                                Gio.FileQueryInfoFlags.NONE, null);
+
+        let prefix = 'eos-folder-user-';
+        let suffix = '.directory';
+        let re = new RegExp(prefix + '([0-9]+)' + suffix);
+        let n = -1;
+
+        for (let f = enumerator.next_file(null); f != null; f = enumerator.next_file(null)) {
+            let result = re.exec(f.get_name());
+            if (result != null) {
+                let new_n = result[1];
+                n = Math.max(n, parseInt(new_n));
+            }
+        }
+        enumerator.close(null);
+
+        let filename = prefix + (n+1) + suffix;
+
+        let buf = keyFile.to_data();
+
+        let channel = GLib.IOChannel.new_file(dirpath + '/' + filename, 'w');
+        channel.write_chars(buf[0], buf[1]);
+        channel.shutdown(true);
+
+        return filename;
+    }
+});
+
+// Use as a singleton
+const model = new FolderModel();
```

Comment:
Instead of making this a singleton, you could simply avoid making a class and call the function directly